src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -31,17 +31,24 @@ export class LocalTaskService implements TaskService {
     }
     return of(null);
   }
-
+  deleteAllDoneTasks(): Observable<void>{
+    const tasks = this.readTasks();
+    const doneTasks = tasks.filter(function(task){
```

Comment:
You can use arrow functions for this which makes easier to write :)
https://basarat.gitbooks.io/typescript/docs/arrow-functions.html